src/pages/edit-account/edit-account.ts

```diff
@@ -67,4 +74,78 @@ export class EditAccountPage {
     });
   }
 
+  /**
+   * Delete user from db
+   *
+   * @param user
+   */
+  public deleteUser() {
+    let alertButtons = [
+      {
+        role: "cancel",
+        text: "Cancel",
+      },
+      {
+        handler: () => {
+          this.handleDelete();
+        },
+        text: "Confirm",
+      },
+    ];
+
+    Alert(this.alertCtrl, "Delete User", "Are you sure you want to delete your account?", alertButtons);
+  }
+
+  /**
+   * Display successful contact deletion confirmation
+   *
+   */
+  private presentToast() {
+    let toast = this.toastCtrl.create({
+      duration: 2500,
+      message: "User was deleted successfully",
+      position: "top",
+    });
+
+    toast.present();
+  }
+
+  /**
+   * Close the modal page.
+   */
+  private dismiss() {
+    this.viewCtrl.dismiss().catch( (f) => f);
+  }
+
+  /**
+   * Handle user deletion.
+   */
+  private handleDelete() {
+    this.auth.deleteUser()
+      .subscribe(
+        (res: Response) => {
+          let response = res.json();
+
+          if (response.status === "ok") {
+            this.auth.logout();
+
+            this.navCtrl.popToRoot();
+            this.navCtrl.push(LoginPage);
+
+            this.dismiss();
+            this.presentToast();
+          } else {
+            let alertButtons = [
+              {
```

Comment:
Why does this need a variable? Can't you just have
Alert(this.alertCtrl, "Delete User Failed", response.errors.message, [{role: "cancel", text: "Dismiss"}]);